documentation/general/design/dash-sonic-hld.md

```diff
@@ -0,0 +1,467 @@
+| 0.1 | 02/01/2022  |     Prince Sunny   | Initial version                   |
+| 0.2 | 03/09/2022  |     Prince Sunny   | Packet Flows/DB Objects           |
+
+
+# About this Manual
+This document provides more detailed design of DASH APIs, DASH orchestration agent, Config and APP DB Schemas and other SONiC buildimage changes required to bring up SONiC image on an appliance card. General DASH HLD can be found at [dash_hld](https://github.com/Azure/DASH/blob/main/documentation/general/design/dash-high-level-design.md).
+
+# Definitions/Abbreviation
+###### Table 1: Abbreviations
+|                          |                                |
+|--------------------------|--------------------------------|
+| DASH                     | Disaggregated APIs for SONiC Hosts |
+| VNI                      | Vxlan Network Identifier       |
+| VTEP                     | Vxlan Tunnel End Point         |
+| VNET                     | Virtual Network                |
+| ENI                      | Elastic Network Interface      |
+| gNMI                     | gRPC Network Management Interface |
+| vPORT                    | VM's NIC. Eni, Vnic, VPort are used interchangeably |
+
+# 1 Requirements Overview
+
+## 1.1 Functional requirements
+
+At a high level the following should be supported:
+  
+- Bringup SONiC image for DEVICE_METADATA subtype - `appliance`
+- Bringup Swss/Syncd containers for switch_type - `dpu`
+- Able to program DASH objects configured via gRPC client to appliance card via SAI DASH API
+
+  Phase 1
+    - Vnet-Vnet scenario
+    - Route/LPM support
+    - Underlay IPv4 and IPv6
+    - Stateful ACL support
+    - TCP state tracking on flows
+    - Telemetry and Monitoring
+
+  Phase 2
+    - Private Link
+    - Service Tunnel
+    - Overlay IPv6 
+  
+
+## 1.2 CLI requirements
+Initial support is only for `show` commands
+
+- User should be able to show the DASH configured objects
+
+## 1.3 Warm Restart requirements
+Warm-restart support is not considered in Phase 1. TBD
+
+## 1.4 Scaling requirements
+Following are the minimal scaling requirements
+| Item                     | Expected value              |
+|--------------------------|-----------------------------|
+| VNETs                    | 1024                     |
+| ENI                      | 32 Per Card              |
+| Routes per ENI           | 100k                     |
+| NSGs per ENI             | 6                        |
+| ACLs per ENI             | 6x100K prefixes          |
+| ACLs per ENI             | 6x10K SRC/DST ports      |
+| CA-PA Mappings           | 10M                      |
+| Active Connections/ENI   | 1M                       |
+
+# 2 Packet Flows
+	
+The following section captures at a high-level on the VNET packet flow. Detailed lookup and pipeline behavior can be referenced *here*.
+
+## 2.1 Outbound packet processing pipeline
+	
+  ![dash-outbound](https://github.com/prsunny/DASH/blob/main/Assets/pkt_flow_outbound.png)
```

Comment:
We recommend editable .svg format, e.g. using draw.io, so that the images can be maintained down the road. Also please use a relative URL, this points to an absolute path in a personal repo dev branch.

Reply:
@prsunny Please, replace the current file image with this: 
![dash-hld-outbound-packet-processing-pipeline](https://user-images.githubusercontent.com/2974753/160506366-e18a3924-3aa2-49bc-b54d-21e92533e059.svg)